services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/events.py

```diff
@@ -503,114 +506,133 @@ async def action(cls, app: FastAPI, scheduler_data: SchedulerData) -> None:
         # invoke container cleanup at this point
         dynamic_sidecar_client: DynamicSidecarClient = get_dynamic_sidecar_client(app)
 
-        try:
-            await dynamic_sidecar_client.begin_service_destruction(
-                dynamic_sidecar_endpoint=scheduler_data.dynamic_sidecar.endpoint
-            )
-        except BaseClientHTTPError as e:
-            logger.warning(
-                (
-                    "Could not contact dynamic-sidecar to begin destruction of "
-                    "%s\n%s. Will continue service removal!"
-                ),
-                scheduler_data.service_name,
-                f"{e}",
-            )
-        app_settings: AppSettings = app.state.settings
-        dynamic_sidecar_settings: DynamicSidecarSettings = (
-            app_settings.DYNAMIC_SERVICES.DYNAMIC_SIDECAR
-        )
-
-        # only try to save the status if :
-        # - the dynamic-sidecar has finished booting correctly
-        # - it is requested to save the state
-        if (
-            scheduler_data.dynamic_sidecar.service_removal_state.can_save
-            and await are_all_services_present(
-                node_uuid=scheduler_data.node_uuid,
-                dynamic_sidecar_settings=app.state.settings.DYNAMIC_SERVICES.DYNAMIC_SIDECAR,
+        # A node can end up with all the services from a single study.
+        # When the study is closed, all the services will try to save their
+        # data. This causes a lot of disk and network stress.
+        # Some nodes collapse under load or behave unexpectedly.
+        lock_manager = RedisLockManager.instance(app)
+        assert scheduler_data.dynamic_sidecar_node_id  # nosec
+        lock: Optional[Lock] = await lock_manager.acquire_lock(
+            scheduler_data.dynamic_sidecar_node_id
+        )
+        if lock is None:
```

Comment:
Suggestion: 
```python
if lock := await lock_manager.acquire_lock...:

Reply:
This will not work. I require the lock distance to release it below.